modules/http4s/src/smithy4s/http4s/SimpleProtocolBuilder.scala

```diff
@@ -91,27 +95,40 @@ abstract class SimpleProtocolBuilder[P](val codecs: CodecAPI)(implicit
       _
   ]: EffectCompat] private[http4s] (
       client: Client[F],
-      service: smithy4s.Service[Alg, Op],
-      uri: Uri = uri"http://localhost:8080"
+      val service: smithy4s.Service[Alg, Op],
+      errorOrUri: Either[InvalidUriError, Uri] = Right(
+        uri"http://localhost:8080"
+      )
   ) {
 
     def uri(uri: Uri): ClientBuilder[Alg, Op, F] =
-      new ClientBuilder[Alg, Op, F](this.client, this.service, uri)
+      new ClientBuilder[Alg, Op, F](this.client, this.service, Right(uri))
+
+    def uri(uri: String): ClientBuilder[Alg, Op, F] =
```

Comment:
This change pollutes the API in a way that breaks bincompat, and I don't think it's worth the breakage.